### PR TITLE
waypaper: init at 1.2

### PR DESCRIPTION
**Pull request by TotalChaos05**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by OPNA2608 — CHANGES_REQUESTED** [review]
Additionally, please switch to the `finalAttrs` pattern if `buildPythonApplication` supports it (not tested): https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes

What it more or less boils down to:
```nix
python3.pkgs.buildPythonApplication rec {
  version = "1.2";

  src = fetchFromGitHub {
    rev = "refs/tags/${version}";
  };

  meta = with lib; {
    changelog = "https://github.com/anufrievroman/waypaper/releases/tag/${version}";
  };
}
```
becomes
```nix
python3.pkgs.buildPythonApplication (finalAttrs: {
  version = "1.2";

  src = fetchFromGitHub {
    rev = "refs/tags/${finalAttrs.version}";
  };

  meta = with lib; {
    changelog = "https://github.com/anufrievroman/waypaper/releases/tag/${finalAttrs.version}";
  };
})
```

**Comment by TotalChaos05** [comment]
should all be fixed now, thanks for the review !

File: pkgs/applications/misc/waypaper/default.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  nativeBuildInputs = [
+    gobject-introspection
+    wrapGAppsHook
+  ];
+
+  propagatedBuildInputs = with python3.pkgs; [
+    pygobject3
+  ];
+  # has no tests
```
Comment (OPNA2608) [inline pkgs/applications/misc/waypaper/default.nix:28]: Spacing.
```suggestion
  ];

  # has no tests
```

File: pkgs/applications/misc/waypaper/default.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    changelog = "https://github.com/anufrievroman/waypaper/releases/tag/${version}";
+    description = "GUI wallpaper setter for Wayland-based window managers";
+    homepage = "https://github.com/anufrievroman/waypaper";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ totalchaos ];
+    longDescription = ''
+    GUI wallpaper setter for Wayland-based window managers that works as a frontend for popular backends like swaybg and swww.
+
+    If wallpaper does not change, make sure that swaybg or swww is installed.
+  '';
```
Comment (OPNA2608) [inline pkgs/applications/misc/waypaper/default.nix:47]: Order (`longDescription` is usually kept right under `description`), indentation.

```suggestion
    description = "GUI wallpaper setter for Wayland-based window managers";
    longDescription = ''
      GUI wallpaper setter for Wayland-based window managers that works as a frontend for popular backends like swaybg and swww.

      If wallpaper does not change, make sure that swaybg or swww is installed.
    '';
    homepage = "https://github.com/anufrievroman/waypaper";
    license = licenses.gpl3Only;
    maintainers = with maintainers; [ totalchaos ];
```

File: pkgs/applications/misc/waypaper/default.nix [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+}:
+
+python3.pkgs.buildPythonApplication (finalAttrs: {
+  pname = "waypaper";
+  version = "1.2";
+
+  src = fetchFromGitHub {
+    owner = "anufrievroman";
+    repo = "waypaper";
+    rev = "refs/tags/${version}";
```
Comment (OPNA2608) [inline pkgs/applications/misc/waypaper/default.nix:16]:
```suggestion
    rev = "refs/tags/${finalAttrs.version}";
```

File: pkgs/applications/misc/waypaper/default.nix [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  doCheck = false;
+
+  dontWrapGApps = true;
+
+  preFixup = ''
+    makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
+  '';
+
+  meta = with lib; {
+    changelog = "https://github.com/anufrievroman/waypaper/releases/tag/${version}";
```
Comment (OPNA2608) [inline pkgs/applications/misc/waypaper/default.nix:38]:
```suggestion
    changelog = "https://github.com/anufrievroman/waypaper/releases/tag/${finalAttrs.version}";
```

**Review by OPNA2608 — CHANGES_REQUESTED** [review]
Oh also, please move the change to `maintainers/maintainer-list.nix` into a separate commit that preceeds the `waypaper: init` one. Check the history of that file for commit title examples.

**Comment by TotalChaos05** [comment]
it should be fixed now, do i need to rename the pr because i added me as a maintainer?

**Comment by OPNA2608** [comment]
No, current name is fine.

File: pkgs/applications/misc/waypaper/default.nix [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    GUI wallpaper setter for Wayland-based window managers that works as a frontend for popular backends like swaybg and swww.
+
+    If wallpaper does not change, make sure that swaybg or swww is installed.
+  '';
+    homepage = "https://github.com/anufrievroman/waypaper";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ totalchaos ];
+  };
+})
+
```
Comment (OPNA2608) [inline pkgs/applications/misc/waypaper/default.nix:51]:
```suggestion
```

**Review by OPNA2608 — CHANGES_REQUESTED** [review]

**Comment by TotalChaos05** [comment]
also, editorconfig said i needed the newline at the end.

File: pkgs/applications/misc/waypaper/default.nix [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+}:
+
+python3.pkgs.buildPythonApplication (finalAttrs: {
+  pname = "waypaper";
+  version = "1.2";
+
+  src = fetchFromGitHub {
+    owner = "anufrievroman";
+    repo = "waypaper";
+    rev = "refs/tags/${finalAttrs.version}";
```
Comment (OPNA2608) [inline pkgs/applications/misc/waypaper/default.nix:16]:
```suggestion
    rev = "refs/tags/${version}";
```

File: pkgs/applications/misc/waypaper/default.nix [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  doCheck = false;
+
+  dontWrapGApps = true;
+
+  preFixup = ''
+    makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
+  '';
+
+  meta = with lib; {
+    changelog = "https://github.com/anufrievroman/waypaper/releases/tag/${finalAttrs.version}";
```
Comment (OPNA2608) [inline pkgs/applications/misc/waypaper/default.nix:39]:
```suggestion
    changelog = "https://github.com/anufrievroman/waypaper/releases/tag/${version}";
```

File: pkgs/applications/misc/waypaper/default.nix [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    longDescription = ''
+      GUI wallpaper setter for Wayland-based window managers that works as a frontend for popular backends like swaybg and swww.
+
+      If wallpaper does not change, make sure that swaybg or swww is installed.
+    '';
+    homepage = "https://github.com/anufrievroman/waypaper";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ totalchaos ];
+  };
+})
```
Comment (OPNA2608) [inline pkgs/applications/misc/waypaper/default.nix:50]:
```suggestion
}
```

**Review by OPNA2608 — CHANGES_REQUESTED** [review]

**Review by OPNA2608 — APPROVED** [review]
Result of `nixpkgs-review pr 248147` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>waypaper</li>
    <li>waypaper.dist</li>
  </ul>
</details>

---

Checked on aarch64-linux under Miriway WM, LGTM.

File: pkgs/applications/misc/waypaper/default.nix [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    description = "GUI wallpaper setter for Wayland-based window managers";
+    longDescription = ''
+      GUI wallpaper setter for Wayland-based window managers that works as a frontend for popular backends like swaybg and swww.
+
+      If wallpaper does not change, make sure that swaybg or swww is installed.
+    '';
+    homepage = "https://github.com/anufrievroman/waypaper";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ totalchaos ];
+  };
```
Comment (OPNA2608) [inline pkgs/applications/misc/waypaper/default.nix:49]: Actually looking at OfBorg attempting to build this on macOS, this should prolly be limited to Linux (even if it builds on Darwin, I don't think there's any use for it on there):
```suggestion
    platforms = platforms.linux;
  };
```

**Review by OPNA2608 — CHANGES_REQUESTED** [review]

**Review by OPNA2608 — APPROVED** [review]
I *think* now we're good.

**Pull request merged**